Include/internal/pycore_runtime.h

```diff
@@ -27,11 +27,6 @@ extern "C" {
 #include "pycore_typeobject.h"      // struct types_runtime_state
 #include "pycore_unicodeobject.h"   // struct _Py_unicode_runtime_ids
 
-struct _getargs_runtime_state {
-    PyThread_type_lock mutex;
```

Comment:
I think having a lock isn't necessary in 3.12 as we're muving this struct to InterpreterState. Local state modification will be protected by the GIL.